File: ACM2/Services/Backend/CopilotClient.swift
import Foundation

private struct CopilotQueryRequest: Codable {
    let userId: String
    let query: String
}

struct CopilotClient {
    private let backend: BackendClient

    init(backend: BackendClient = BackendClient()) {
        self.backend = backend
    }

    func fetchDailyBrief(userId: String) async throws -> DailyBrief {
        try await backend.get("/copilot/daily-brief", queryItems: [
            URLQueryItem(name: "userId", value: userId)
        ])
    }

    func query(userId: String, query: String) async throws -> CopilotQueryResponse {
        try await backend.post("/copilot/query", body: CopilotQueryRequest(userId: userId, query: query))
    }
}

File: mobile/src/services/backend/CopilotClient.ts
import { CopilotQueryApiResponse, CopilotQueryRequest, DailyBriefResponse } from "./contracts";
import { BackendClient } from "./BackendClient";

export class CopilotClient {
  constructor(private readonly backend = new BackendClient()) {}

  getDailyBrief(): Promise<DailyBriefResponse> {
    return this.backend.get<DailyBriefResponse>("/copilot/daily-brief");
  }

  query(request: CopilotQueryRequest): Promise<CopilotQueryApiResponse> {
    return this.backend.post<CopilotQueryApiResponse>("/copilot/query", request);
  }
}
